Mathlib/RingTheory/MvPolynomial/Homogeneous.lean

```diff
@@ -114,23 +110,20 @@ variable [CommSemiring R]
 
 theorem isHomogeneous_monomial (d : σ →₀ ℕ) (r : R) (n : ℕ) (hn : ∑ i in d.support, d i = n) :
     IsHomogeneous (monomial d r) n := by
-  intro c hc
-  classical
-  rw [coeff_monomial] at hc
-  split_ifs at hc with h
-  · subst c
-    exact hn
-  · contradiction
+  simp_rw [degree_eq_weightedDegree] at hn
+  exact isWeightedHomogeneous_monomial 1 d r hn
 #align mv_polynomial.is_homogeneous_monomial MvPolynomial.isHomogeneous_monomial
 
 variable (σ)
 
+theorem isHomogeneous_of_totalDegree_zero_iff {p : MvPolynomial σ R} :
+    p.totalDegree = 0 ↔ IsHomogeneous p 0 := by
+  rw [totalDegree_eq_weightedTotalDegree,
+    ← isWeightedHomogeneous_zero_iff_weightedTotalDegree_eq_zero, IsHomogeneous]
+
 theorem isHomogeneous_of_totalDegree_zero {p : MvPolynomial σ R} (hp : p.totalDegree = 0) :
-    IsHomogeneous p 0 := by
-  erw [totalDegree, Finset.sup_eq_bot_iff] at hp
-  -- we have to do this in two steps to stop simp changing bot to zero
-  simp_rw [mem_support_iff] at hp
-  exact hp
+    IsHomogeneous p 0 :=
+  (isHomogeneous_of_totalDegree_zero_iff _).mp hp
```

Comment:
You can save a bit of work by using the `alias` command to derive this lemma from `totalDegree_zero_iff_isHomogeneous`.

Reply:
What do you mean?

Reply:
It's been a while so maybe I forgot some important context. It seems that we can just define this as:
```
alias ⟨isHomogeneous_of_totalDegree_zero, _⟩ := isHomogeneous_of_totalDegree_zero_iff
```